crates/red_knot_test/src/assertion.rs

```diff
@@ -292,6 +319,12 @@ pub(crate) struct ErrorAssertion<'a> {
     pub(crate) message_contains: Option<&'a str>,
 }
 
+impl<'a> ErrorAssertion<'a> {
+    fn from_str(source: &'a str) -> Result<Self, ErrorAssertionParseError<'a>> {
+        ErrorAssertionParser::new(source).parse()
+    }
+}
```

Comment:
nit: you could implement [`FromStr`](https://doc.rust-lang.org/stable/std/str/trait.FromStr.html) instead